src/components/Header/Header.tsx

```diff
@@ -0,0 +1,58 @@
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[];
+  query: string;
+  setQuery: React.Dispatch<React.SetStateAction<string>>;
+  addTodo: () => Promise<void>;
+  disabled: boolean;
+  inputRef: React.RefObject<HTMLInputElement>;
+  patch: (event: React.MouseEvent<HTMLButtonElement>) => Promise<void>;
+};
+
+export const Header: React.FC<Props> = ({
+  todos,
+  query,
+  setQuery,
+  addTodo,
+  disabled,
+  inputRef,
+  patch,
+}) => {
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    setQuery(event.target.value);
+  };
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    await addTodo();
+  };
+
+  return (
+    <header className="todoapp__header">
+      {todos.length > 0 && (
+        <button
+          type="button"
+          className={`todoapp__toggle-all${todos.every(todo => todo.completed === true) ? ' active' : ''}`}
```

Comment:
Also, move this logic to the helper variable ans use classnames
```suggestion
          className={`todoapp__toggle-all${todos.every(todo => todo.completed) ? ' active' : ''}`}
```